rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,148 @@
+
+## Problem statement
+
+<!--
+Why are we making this change? What problem are we solving? What do we expect to gain from this?
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+Currently converged package contains following files:
+
+> We'll use contrived name of package - `example` which lives in `/packages/example` folder
+
+- `index.ts` - barrel / Public API surface
+- `Example.tsx` - (complete component to be used as is)
+- `useExample.ts` - (CONTROLLER/MODEL /main logic behind component - re-usable hook)
+- `useExampleClasses.ts` - (STYLES/ main logic for styling component - re-usable hook)
+- `useExampleRandom.tsx` - (some encapsulated logic - not really a hook)
+- `renderExample.tsx` - (VIEW/ main logic that handles rendering - re-usable hook)
+- `Example.types.ts` - (TYPES/ all TypeScript types used within package)
+
+**Potential issues from developer point of view:**
+
+- inconsistent file naming
+  - some files contain suffix (`.types`)
+  - some files have `.ts` some `.tsx` what's the difference ?
+  - some files are PascalCased, some camelCased
+- not explicitly communicating the intent
+  - what does `use` mean ?
+    - seasoned react dev will probably guess that it's a hook. Thing is, that this file might export more that 1 hook.
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+- use consistent names for all files.
+- follow a pattern that describes the symbol's feature or the file scope(component name) then its type.
+  - The recommended pattern is `feature.type.ts`.
+- use conventional **type** names including
+  - `.hook`
+  - `.view`
+  - `.styles`
+  - `.types`
+  - `.utils`
+  - `.stories`
+  - invent additional type names if you must but take care not to create too many
+
+**Why?**
+
+- Type names provide a consistent way to quickly identify what is in the file.
+- Type names make it easy to find a specific file type using an editor or IDE's fuzzy search techniques.
+- Type names provide pattern matching for any automated tasks.
+
+---
+
+- use `.tsx` only for files that contain view/rendering related logic
+
+**Why?**
+
+- will enforce to use JSX only where necessary (renderer only)
+- same as for Type Names
+
+---
+
+- (optional) use `.spec.` instead `.test.` suffix as we already follow behavioral driven approach (describes/it) rather than atomic `test()` approach
+  - as a next step, enable lint rule that will enforce one way or another within tests
+
+**Why?**
+
+- consistency/proper intent
+
+---
+
+- (optional) use kebab case for naming
+
+**Why?**
+
+- kebab case is more human readable especially in longer words
```

Comment:
Either kebab or snake, but agreed that either is superior to camel or Pascal case for file names.